Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4800,6 +4792,32 @@ - (void)extendRangeToIncludeUnobservedItems
     self.previousLastTimestamp = nil;
 }
 
+- (void)createNewMessageMappings
+{
+    if (self.thread.uniqueId.length > 0) {
+        self.messageMappings = [[YapDatabaseViewMappings alloc] initWithGroups:@[ self.thread.uniqueId ]
+                                                                          view:TSMessageDatabaseViewExtensionName];
+    } else {
+        OWSFail(@"uniqueId unexpectedly empty for thread: %@", self.thread);
+        self.messageMappings =
+            [[YapDatabaseViewMappings alloc] initWithGroups:@[] view:TSMessageDatabaseViewExtensionName];
+    }
+
+    // Cells' appearance can depend on adjacent cells in both directions.
+    [self.messageMappings setCellDrawingDependencyOffsets:[NSSet setWithArray:@[
+        @(-1),
+        @(+1),
+    ]]
+                                                 forGroup:self.thread.uniqueId];
+
+    // We need to impose the range restrictions on the mappings immediately to avoid
```

Comment:
seems like this comment applies to the code 3 lines down.

Reply:
Done.